rfcs/20240326-block-level-api/README.md

```diff
@@ -0,0 +1,540 @@
+# Proposal for block level APIs
```

Comment:
Since we are exposing this internal API. 
Are there any restrictions on modifying? If there are, how should the changes in brgemm be communicated?
Ideally, we wish it to be quicker than RFC. Or is it possible to isolate internal from external somehow.

Reply:
There shouldn't be any implications on internals. This is a new C layer on top of existing one with a limited set of arguments. Internal C layer will be re-worked into proper C++ eventually. Changes to internal C API will be an implementation detail of external one and should be updated accordingly to keep the functionality working.